docs/modules/spark-k8s/pages/usage-guide/resources.adoc

```diff
@@ -50,4 +50,151 @@ For more details regarding Kubernetes CPU limits see: https://kubernetes.io/docs
 
 Spark allocates a default amount of non-heap memory based on the type of job (JVM or non-JVM). This is taken into account when defining memory settings based exclusively on the resource limits, so that the "declared" value is the actual total value (i.e. including memory overhead). This may result in minor deviations from the stated resource value due to rounding differences.
 
-NOTE: It is possible to define Spark resources either directly by setting configuration properties listed under `sparkConf`, or by using resource limits. If both are used, then `sparkConf` properties take precedence. It is recommended for the sake of clarity to use *_either_* one *_or_* the other.
+NOTE: It is possible to define Spark resources either directly by setting configuration properties listed under `sparkConf`, or by using resource limits. If both are used, then `sparkConf` properties take precedence. It is recommended for the sake of clarity to use *_either_* one *_or_* the other. See below for examples.
+
+== Resource examples
+
+To illustrate resource configuration consider the use-case where resources are defined using CRD fields (which are then parsed internally to be passed to Spark as spark.conf settings).
+
+=== CPU
+
+CPU/Cores will be rounded up and only applied as a limit (instead of separately as request and limit). The rounding to the next integer value will result in the following:
+
+
+|===
+|CRD |Spark conf
+
+|1800m
+|2
+
+|100m
+|1
+
+|1.5
+|2
+
+|2
+|2
+|===
+
+As each `SparkApplication` will be used to derive templates for the driver and executor pods (pods which are eventually terminated once the job is complete) it is not so crucial to set a resource request separately: the resource limit will be applied to `spark.conf` settings such that the maximum value is applied for the duration of the application. This means in effect that `config.resources.cpu.min` can be ignored by the user, although the field has not been removed to avoid any possible future breaking changes.
```

Comment:
As a user I would be surprised that my request gets overwritten by the limit. Is their any specific reason for this or can we fix this?

Reply:
It is because we use a common struct and `request` for memory does not map onto any spark-conf setting.

Reply:
Ok, maybe I need to re-phrase a bit ^^
Why don't we just pass CPU request and limit to the Spark executors? For memory request is the same limits and that's totally fine.

Reply:
Sorry, mis-read your comment. I'm not sure why we don't pass through the cores request. We currently set all these

- spark.{driver,executor}.cores
- spark.kubernetes.{driver,executor}.request.cores
- spark.kubernetes.{driver,executor}.limit.cores

with the same value (cpu/max) which doesn't make sense as the second one on that list has precedence over the first one.

Reply:
Yeah, I have used different spark.kubernetes.executor.request.cores and spark.kubernetes.executor.limit.cores configs in production in the past, so I think we should fix that if possible

Reply:
Agreed. Am on it!

Reply:
Awesome, thanks!

Reply:
Changed in db89289

Reply:
Sorry to bother you, but we still don't pass the request (and probably limit) through: The users specifies 250m request, we quadruple it to 1000m

Reply:
That's because we are rounding it up, which I have tried to reflect in the docs: "...The rounding to the next integer value will result in the following...". Should that be made clearer e.g. "...to the next whole number of cores.."?

Reply:
Yeah, my point is: why we don't pass through what the users specifies? Why is there a rounding in the first place?

Reply:
We decided to keep this a documentation PR and discuss the mentioned points as follow-ups